mmdet3d/datasets/dataset_wrappers.py

```diff
@@ -0,0 +1,78 @@
+import mmcv
+import numpy as np
+
+from .builder import DATASETS
+
+
+@DATASETS.register_module()
+class ClassSampledDataset(object):
+    """A wrapper of class sampled dataset with ann_file path.
+
+    Balance the number of scenes under different classes.
+
+    Args:
+        dataset (:obj:`CustomDataset`): The dataset to be class sampled.
+        ann_file (str): Path of annotation file.
+    """
+
+    def __init__(self, dataset, ann_file):
+        self.dataset = dataset
+        self.CLASSES = dataset.CLASSES
+        self.repeat_indices = self._get_repeat_indices(ann_file)
+        # self.dataset.data_infos = self.data_infos
+        if hasattr(self.dataset, 'flag'):
+            self.flag = np.array(
+                [self.dataset.flag[ind] for ind in self.repeat_indices],
+                dtype=np.uint8)
+
+    def _get_repeat_indices(self, ann_file):
```

Comment:
`repeat_indices` -> `sample_indices` since you only sample rather than repeat the data.